### ISSUE TITLE
New-AzBotService requires -Location, while New-AzResourceGroupDeployment does not have -Location

### ISSUE THREAD

**Issue by jg-cloudzbridge** [Azure/azure-powershell#17185]
### Description

New-AzBotService -ResourceGroupName "Vision" -Name "SomeBot" -ApplicationId "DUMMY ID" -location global -ApplicationSecret $d -Sku S0 -Webapp
New-AzResourceGroupDeployment : A parameter cannot be found that matches parameter name 'Location'.
At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:242 char:93
+ ... oupDeployment -ResourceGroupName $ResourceGroupName -Location $Locati ...
+                                                         ~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-AzResourceGroupDeployment], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.N
   ewAzureResourceGroupDeploymentCmdlet

See https://www.powershellgallery.com/packages/Az.BotService/0.3.2/Content/custom%5CNew-AzBotService.ps1 line 243
See https://docs.microsoft.com/en-us/powershell/module/az.botservice/new-azbotservice?view=azps-7.2.0

### Issue script & Debug output

```PowerShell
DEBUG: 11:23:13 AM - GetAzureRMContextCommand end processing.
DEBUG: [CmdletBeginProcessing]: Starting command
DEBUG: CmdletBeginProcessing:
DEBUG: CmdletProcessRecordStart:
DEBUG: [CmdletProcessRecordAsyncStart]: Created new QosEvent for command 'Get-AzBotCheckNameAvailability_GetExpanded'
DEBUG: CmdletProcessRecordAsyncStart:
DEBUG: CmdletGetPipeline:
DEBUG: CmdletBeforeAPICall:
DEBUG: URLCreated: /providers/Microsoft.BotService/checkNameAvailability?api-version=2018-07-12
DEBUG: RequestCreated: /providers/Microsoft.BotService/checkNameAvailability?api-version=2018-07-12
DEBUG: HeaderParametersAdded:
DEBUG: BodyContentSet:
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
POST

Absolute Uri:
https://management.azure.com/providers/Microsoft.BotService/checkNameAvailability?api-version=2018-07-12

Headers:
x-ms-unique-id                : 22
x-ms-client-request-id        : "DUMMY UUID"
CommandName                   : Az.BotService.internal\Get-AzBotCheckNameAvailability
FullCommandName               : Get-AzBotCheckNameAvailability_GetExpanded
ParameterSetName              : __AllParameterSets
User-Agent                    : AzurePowershell/v0.0.0,PSVersion/v5.1.17763.1852,Az.BotService/0.3.2

Body:
{
  "name": "NPOBot",
  "type": "sdk"
}

DEBUG: BeforeCall:
DEBUG: Finally:
DEBUG: [CmdletProcessRecordAsyncEnd]: Finish HTTP process
DEBUG: CmdletProcessRecordAsyncEnd:
DEBUG: [CmdletException]: Received Exception with message 'HttpRequestException - An error occurred while sending the
request. :    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Commands.Common.ContextAdapter.<>c__DisplayClass13_0.<<SendHandler>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability_Call>d__39.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability_Call>d__39.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at
Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetExpanded.<ProcessRecordAsync>d__
62.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at
Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetExpanded.<ProcessRecordAsync>d__
62.MoveNext()'
DEBUG: CmdletException: HttpRequestException - An error occurred while sending the request. :    at
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Commands.Common.ContextAdapter.<>c__DisplayClass13_0.<<SendHandler>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability_Call>d__39.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability_Call>d__39.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at
Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetExpanded.<ProcessRecordAsync>d__
62.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at
Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetExpanded.<ProcessRecordAsync>d__
62.MoveNext()
Az.BotService.internal\Get-AzBotCheckNameAvailability : An error occurred while sending the request.
At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:193 char:13
+             $NameAvailabilityResponse = Az.BotService.internal\Get-Az ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-AzBotCheckN...ity_GetExpanded], HttpRequestException
    + FullyQualifiedErrorId : Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetE
   xpanded

DEBUG: CmdletProcessRecordEnd:
DEBUG: AzureQoSEvent: Module: Az.BotService:0.3.2; CommandName: Get-AzBotCheckNameAvailability_GetExpanded; PSVersion:
5.1.17763.1852; IsSuccess: True; Duration: 00:00:00
DEBUG: Finish sending metric.
DEBUG: CmdletEndProcessing:
New-AzBotService : Exception of type 'Microsoft.PowerShell.Commands.WriteErrorException' was thrown.
At line:1 char:1
+ New-AzBotService -ResourceGroupName "Vision" -Name "NPOBot" -Applicat ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,New-AzBotService

ScriptHalted
At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:197 char:17
+                 throw
+                 ~~~~~
    + CategoryInfo          : OperationStopped: (:) [], RuntimeException
    + FullyQualifiedErrorId : ScriptHalted
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      5.1.17763.1852
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.1852
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Module versions

```PowerShell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     2.7.2      Az.Accounts                         {Add-AzEnvironment, Clear-AzContext, Clear-AzDefault, Conn...
Script     0.3.2      Az.BotService                       {Export-AzBotServiceApp, Get-AzBotService, Initialize-AzBo...
Script     1.7.1      Az.Resources                        {Add-AzADGroupMember, Export-AzResourceGroup, Get-AzADAppC...
```

### Error output

```PowerShell
HistoryId: -1


Message        : ScriptHalted
StackTrace     : 
Exception      : System.Management.Automation.RuntimeException
InvocationInfo : {}
Line           :                 throw
                 
Position       : At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:197 char:17
                 +                 throw
                 +                 ~~~~~
HistoryId      : -1



   HistoryId: 71


Message        : Exception of type 'Microsoft.PowerShell.Commands.WriteErrorException' was thrown.
StackTrace     : 
Exception      : Microsoft.PowerShell.Commands.WriteErrorException
InvocationInfo : {New-AzBotService}
Line           : New-AzBotService -ResourceGroupName "Vision" -Name "NPOBot" -ApplicationId $c -location global -ApplicationSecret $d -Sku 
                 F0 -Webapp
Position       : At line:1 char:1
                 + New-AzBotService -ResourceGroupName "Vision" -Name "NPOBot" -Applicat ...
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 71

Message        : An error occurred while sending the request.
StackTrace     :    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                    at Microsoft.Azure.Commands.Common.ContextAdapter.<>c__DisplayClass13_0.<<SendHandler>b__0>d.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                    at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability_Call>d__39.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability_Call>d__39.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                    at Microsoft.Azure.PowerShell.Cmdlets.BotService.BotService.<BotsGetCheckNameAvailability>d__37.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                    at Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetExpanded.<ProcessRecordAsync>d__62.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at Microsoft.Azure.PowerShell.Cmdlets.BotService.Cmdlets.GetAzBotCheckNameAvailability_GetExpanded.<ProcessRecordAsync>d__62.MoveNext()
Exception      : System.Net.Http.HttpRequestException
InvocationInfo : {Get-AzBotCheckNameAvailability_GetExpanded}
Line           :             $NameAvailabilityResponse = Az.BotService.internal\Get-AzBotCheckNameAvailability -Name $Name -Type $Kind @EnvPSBoundParameters
                 
Position       : At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:193 char:13
                 +             $NameAvailabilityResponse = Az.BotService.internal\Get-Az ...
                 +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 71

Message        : Unable to connect to the remote server
StackTrace     :    at System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
                    at System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
Exception      : System.Net.WebException
InvocationInfo : {Get-AzBotCheckNameAvailability_GetExpanded}
Line           :             $NameAvailabilityResponse = Az.BotService.internal\Get-AzBotCheckNameAvailability -Name $Name -Type $Kind @EnvPSBoundParameters
                 
Position       : At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:193 char:13
                 +             $NameAvailabilityResponse = Az.BotService.internal\Get-Az ...
                 +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 71

Message        : A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected 
                 host has failed to respond 40.79.180.0:443
StackTrace     :    at System.Net.Sockets.Socket.InternalEndConnect(IAsyncResult asyncResult)
                    at System.Net.Sockets.Socket.EndConnect(IAsyncResult asyncResult)
                    at System.Net.ServicePoint.ConnectSocketInternal(Boolean connectFailure, Socket s4, Socket s6, Socket& socket, IPAddress& address, ConnectSocketState 
                 state, IAsyncResult asyncResult, Exception& exception)
Exception      : System.Net.Sockets.SocketException
InvocationInfo : {Get-AzBotCheckNameAvailability_GetExpanded}
Line           :             $NameAvailabilityResponse = Az.BotService.internal\Get-AzBotCheckNameAvailability -Name $Name -Type $Kind @EnvPSBoundParameters
                 
Position       : At C:\Program Files\WindowsPowerShell\Modules\Az.BotService\0.3.2\custom\New-AzBotService.ps1:193 char:13
                 +             $NameAvailabilityResponse = Az.BotService.internal\Get-Az ...
                 +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 71
```

**Comment by dingmeng-xue** [comment]
We will look into it.

**Comment by wyunchi-ms** [comment]
The PR is merged. This will be fixed in the next release.

**Comment by VeryEarly** [comment]
fixed in #18324
